### PR TITLE
view data

### PR DESCRIPTION
**Pull request by turadg**
fixes: #2 

I went a little farther since I depend on this tool so much. E.g. converting to Vite from CRA which is no deprecated.

The main UX change is making data render and doing it a little more clearly when it does.

*Before*

https://vstorage.agoric.net/?path=published.wallet.agoric10yw7lsgqgajcc4veeeua65je79tyeg9x8t6f5y.current&endpoint=https%3A%2F%2Fmain-a.rpc.agoric.net%3A443&height=null

(manually collapsing body children)
Screenshot 2024-12-10 at 4 20 09 PM

**After**

http://localhost:5173/?path=published.wallet.agoric10yw7lsgqgajcc4veeeua65je79tyeg9x8t6f5y.current&endpoint=https%3A%2F%2Fmain-a.rpc.agoric.net%3A443&height=null

(same collapsing)

Screenshot 2024-12-10 at 4 21 00 PM

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -166,7 +165,9 @@ const App = () => {
     updateQueryParam("endpoint", newEndpoint);
   };
 
-  const dataToShow = columns.at(-1).items.length === 0 && dataView;
+  // const dataToShow = columns.at(-1).items.length === 0 && dataView;
+  // console.log("dataToShow", dataToShow, "for", dataView);
+  console.log("dataView", dataView);
```
Comment (0xpatrickdev) [inline src/App.js]: artifact or intentional?

File: src/App.jsx [hunk 2]
```diff
@@ -91,21 +92,29 @@ const App = () => {
...
-        console.log("Data received from fetchData:", response.data);
-        setDataView(response.data.value);
+        console.log("Data received from fetchData:", response.data.value);
+        const take2 = storageHelper.unserializeTxt(response.data.value);
+        console.log("DEBUG take2", take2);
+        const obj = JSON.parse(response.data.value);
+        console.log("DEBUG obj", typeof obj, obj);
+        obj.values = obj.values.map((v) => JSON.parse(v));
+        console.log("DEBUG obj.values", obj.values);
+        setDataView(obj);
```
Comment (0xpatrickdev) [inline src/App.jsx]: Are these console.log's intentional? What's difficult to work out?

I pulled this down locally. For `fastUsdc` it doesn't seem to render keys that include JSON - only capData. Examples are the `fastUsdc` root  and `fastUsdc.status.${hash}`. I also noticed the old `p2p-agoric-vstorage-viewer` struggled with JSON as well.

**Review by 0xpatrickdev — APPROVED** [review]
Pulled down locally to verify.

I see `published.wallet.${addr}` _data_ now rendering which is an improvement.

Didn't audit the parsing logic closely, but left feedback where it seems we're always expecting `capData` and sometimes it's just JSON.

File: src/App.jsx (hunk 2 above)
Comment (turadg) [reply src/App.jsx]: They were intentional while debugging. Looks like you were reviewing by commit. They're not in HEAD.

I don't see much value in cleaning those out of the commit history but would entertain the request

Comment (turadg) [reply src/App.jsx]: I did it tho

Comment (0xpatrickdev) [reply src/App.jsx]: > Looks like you were reviewing by commit

Yep, thanks for removing, though.

> it doesn't seem to render keys that include JSON - only capData

Was this an observation you also had?

Comment (turadg) [reply src/App.jsx]: yes. I want to solve this in client-utils `vstorage.js` or `vstorage-kit.js` by making them handle non-capdata JSON and even bare values

**Pull request merged**